packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -466,17 +466,64 @@ export class OdspDocumentStorageService implements IDocumentStorageService {
         const { headers, url } = getUrlAndHeadersWithAuth(`${this.snapshotUrl}/trees/latest${snapshotOptions}`, storageToken);
 
+        let isOptionsCall = false;
+
+        if (headers.Authorization) {
+            isOptionsCall = true;
+        }
         // This event measures only successful cases of getLatest call (no tokens, no retries).
         const { snapshot, canCache } = await PerformanceEvent.timedExecAsync(this.logger, { eventName: "TreesLatest" }, async (event) => {
+            const startTime = performance.now();
             const response = await fetchHelper<IOdspSnapshot>(url, { headers });
+            const endTime = performance.now();
+            const overallTime = endTime - startTime;
             const content = response.content;
+            let dnstime = -1; // domainLookupEnd - domainLookupStart
+            let redirectTime = -1; // redirectEnd -redirectStart
+            let tcpHandshakeTime = -1; // connectEnd  - connectStart
+            let secureConntime = -1; // connectEnd  - secureConnectionStart
+            let responseTime = -1; // responsEnd - responseStart
+            let fetchStToRespEndTime = -1; // responseEnd  - fetchStart
+            let reqStToRespEndTime = -1; // responseEnd - requestStart
+            let networkTime = -1; // responseEnd - startTime
+
+            const resources1 = performance.getEntriesByType("resource");
+            // Usually the latest fetch call is to the end of resources, so we start from the end.
+            for (let i = resources1.length - 1; i > 0; i--) {
+                const indResTime = resources1[i] as PerformanceResourceTiming;
+                const resource_name = indResTime.name;
+                const resource_initiatortype = indResTime.initiatorType;
+                if ((resource_initiatortype.localeCompare("fetch") === 0) && (resource_name.localeCompare(url) === 0)) {
+                        redirectTime = indResTime.redirectEnd - indResTime.redirectStart;
+                        dnstime = indResTime.domainLookupEnd - indResTime.domainLookupStart;
+                        tcpHandshakeTime = indResTime.connectEnd - indResTime.connectStart;
+                        secureConntime = (indResTime.secureConnectionStart > 0) ? (indResTime.connectEnd - indResTime.secureConnectionStart) : 0;
+                        responseTime = indResTime.responseEnd - indResTime.responseStart;
+                        fetchStToRespEndTime = (indResTime.fetchStart > 0) ? (indResTime.responseEnd - indResTime.fetchStart) : 0;
+                        reqStToRespEndTime = (indResTime.requestStart > 0) ? (indResTime.responseEnd - indResTime.requestStart) : 0;
+                        networkTime = (indResTime.startTime > 0) ? (indResTime.responseEnd - indResTime.startTime) : 0;
+                        break;
+                }
+            }
+            const clientTime = overallTime - networkTime;
```

Comment:
clientTime = overallTime - networkTime - spRequestDuration 
(need to subtract service duration as well).

Reply:
As discussed offline, added a check to make sure to subtract spRequestDuration when it is available.